### PR TITLE
feature/typescript

### PR DESCRIPTION
**Pull request by builtbywill**

### CODE REVIEW

File: src/noStoreSaga.test.ts [hunk 1]
```diff
@@ -177,46 +161,24 @@ describe('helpers', () => {
...
-		const hook = data => {
-			let foo = 1
-		}
-		registerNoStoreActionHook('key', hook)
-		expect(hooks['key']).toEqual(hook)
-		unregisterNoStoreActionHook('key')
-		expect(hooks['key']).toEqual(undefined)
-	})
-})
+// describe('noStoreSaga', () => {
```
Comment (gavin-brown) [inline src/noStoreSaga.test.ts]: Commented out code that could potentially be removed?

Comment (builtbywill) [reply src/noStoreSaga.test.ts]: @gavin-brown thanks! removed

File: src/fetchSaga.ts [hunk 2]
```diff
@@ -480,37 +485,39 @@ function* fetchDataRecurring(action: FetchAction) {
...
  *
  * @export
- * @param {Object} modelsParam - An object indicating the APIs available in a application with which to make requests
- * @param {string} apiRootParam - A url to which partial URLs are appended (i.e.) 'https://myapp.com'
- * @param {TokenAccessFunction} [tokenAccessParam=defaultTokenAccessFunction] - function that returns an optional OAuth token
- * @param {ErrorFunction} [errorParam=defaultErrorFunction]  - A function to perform on errors
- * @param {LoggerFunction} [loggerParam=defaultLogger] - A function that accepts a string and logs it real good
+ * @param {EndpointMappings} endpointMappingsParam An mapping of API endpoints available in the application
+ * @param {string | undefined} apiRootParam A url to which partial URLs are appended (i.e.) 'https://myapp.com'
+ * @param {TokenAccessFunction} [tokenAccessParam=defaultTokenAccessFunction] function that returns
```
Comment (tragessere) [inline src/fetchSaga.ts]: `tokenAccessParam` and `errorParam` jsdoc names don't match the function parameter names `tokenAccessFunctionParam` and `errorFunctionParam`

File: src/fetchReducer.test.ts [hunk 3]
```diff
@@ -1,10 +1,8 @@
-import fetchReducer, { __RewireAPI__ as FetchReducerRewireAPI } from '../src/fetchReducer'
-import actions from '../src/actions'
+import fetchReducer, { getMetadata, isCollection, mergeRelations } from './fetchReducer'
+import actions from './actions'
 import _ from 'lodash'
```
Comment (tragessere) [inline src/fetchReducer.test.ts]: unused import

File: src/noStoreSaga.test.ts [hunk 4]
```diff
@@ -1,39 +1,17 @@
...
-	fork,
-	put,
-	race,
-	select
-} from 'redux-saga/effects'
-import uuid from 'uuid'
+import actions from './actions'
+import { takeEvery } from 'redux-saga/effects'
+import * as uuid from 'uuid'
 import noStoreSaga, {
```
Comment (tragessere) [inline src/noStoreSaga.test.ts]: looks like the default export `noStoreSaga` isn't used either

**Comment by builtbywill** [comment]
@tragessere resolved your issues

File: package.json [hunk 5]
```diff
@@ -2,65 +2,84 @@
...
-		"eslint-config-prettier": "^4.0.0",
-		"eslint-config-studiokit": "^1.0.0",
-		"eslint-plugin-prettier": "^3.0.1",
-		"eslint-plugin-redux-saga": "^1.0.0",
-		"flow-bin": "^0.92.1",
 		"husky": "^1.3.1",
-		"jest": "23.6.0",
-		"lint-staged": "^8.1.3",
-		"prettier": "^1.16.4"
+		"jest": "^24.3.0",
```
Comment (lily514) [inline package.json]: @builtbywill Do we care about these jest dependency warnings?
image

Comment (builtbywill) [reply package.json]: @lily514 I don't think so? seems like a thing that `jest` should be responsible for fixing

File: src/fetchSaga.ts [hunk 6]
```diff
@@ -1,50 +1,50 @@
...
+	FetchError,
+	OAuthToken
+} from './types'
 
 //#region Types
 
-type LoggerFunction = string => void
-type TokenAccessFunction = void => ?OAuthToken
-type ErrorFunction = string => void
+type LoggerFunction = (message: any) => void
```
Comment (lily514) [inline src/fetchSaga.ts]: @builtbywill why is this `(message: any)` instead of `(message: string)`?

Comment (builtbywill) [reply src/fetchSaga.ts]: @lily514 hm, well `console.debug` uses `message?: any. maybe we should use type instead?

File: src/fetchSaga.ts [hunk 7]
```diff
@@ -60,23 +60,27 @@ const defaultLogger: LoggerFunction = (message: string) => {
...
+//#endregion Local Variables
 
 /**
  * Prepare fetchConfig to pass to fetchService. Also set up state
  * to handle response correctly.
  *
- * @param {Object} model - The model selected from the models object
- * @param {FetchAction} action - The action dispatched by the client
- * @param {Object} models - The entire models object, passed in for testability
+ * @param {EndpointMapping} endpointMapping The model selected from the models object
```
[the remark this answers is not in the document]
Comment (builtbywill) [reply src/fetchSaga.ts]: @lily514 updated

File: src/fetchSaga.ts (hunk 6 above)
Comment (lily514) [reply src/fetchSaga.ts]: @builtbywill Oh I see. Um, do you think it matters? No strong opinions here.

File: src/fetchSaga.test.ts [hunk 8]
```diff
@@ -1,34 +1,23 @@
...
+	fetchDataLoop,
+	fetchDataRecurring,
+	fetchOnce,
+	getState,
+	matchesTerminationAction,
+	prepareFetch,
+	takeMatchesTerminationAction
+} from './fetchSaga'
+import { doFetch } from './fetchService'
+import { OAuthToken, OAuthTokenResponse, TokenAccessFunction } from './types'
```
Comment (tragessere) [inline src/fetchSaga.test.ts]: OAuthTokenResponse isn't used anymore

**Pull request merged**